tools/esp32c3/Config.mk

```
@@ -52,13 +52,12 @@ else ifeq ($(CONFIG_ESP32C3_FLASH_FREQ_20M),y)
 	FLASH_FREQ := 20m
 endif
 
-ESPTOOL_FLASH_OPTS := -fs $(FLASH_SIZE) -fm $(FLASH_MODE) -ff $(FLASH_FREQ)
-ESPTOOL_ELF2IMG_OPTS := $(ESPTOOL_FLASH_OPTS)
+ESPTOOL_ELF2IMG_OPTS := -fs $(FLASH_SIZE) -fm $(FLASH_MODE) -ff $(FLASH_FREQ)
 
 ifeq ($(CONFIG_ESP32C3_FLASH_DETECT),y)
-	ESPTOOL_WRITEFLASH_OPTS := -fs detect -fm $(FLASH_MODE) -ff $(FLASH_FREQ)
+	ESPTOOL_WRITEFLASH_OPTS := -fs detect -fm dio -ff $(FLASH_FREQ)
 else
-	ESPTOOL_WRITEFLASH_OPTS := $(ESPTOOL_FLASH_OPTS)
+	ESPTOOL_WRITEFLASH_OPTS := -fs detect -fm dio -ff $(FLASH_FREQ)
```

```suggestion
	ESPTOOL_WRITEFLASH_OPTS := -fs $(FLASH_SIZE) -fm dio -ff $(FLASH_FREQ)
```
Here I think you forgot the FLASH_SIZE param

---

That's right.  Thanks.